bee-trace/bee-trace/examples/flamegraph.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use bee_trace::observe;
+
+use tokio::{task, time::timeout};
+
+use std::{path::PathBuf, time::Duration};
+
+#[tokio::main(flavor = "current_thread")]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let examples_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("examples");
+
+    let stack_filename = examples_dir.clone().join("trace");
+
+    println!(
+        "Creating flamegraph layer, recording to {}.folded",
+        stack_filename.to_string_lossy()
+    );
+
+    let flamegrapher = bee_trace::subscriber::build()
+        .with_flamegraph_layer(stack_filename)
+        .init()?
+        .unwrap();
+
+    println!("Running tasks for 5 seconds...");
+    let task1 = task::spawn(timeout(Duration::from_secs(5), run_tasks(500)));
+    let task2 = task::spawn(timeout(Duration::from_secs(5), run_tasks(200)));
+    let _ = tokio::join!(task1, task2);
+
+    let graph_filename = examples_dir.clone().join("flamegraph");
```

Comment:
I think you can avoid this `.clone()` as `examples_dir` is not used anywhere else after this.